src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -715,6 +715,27 @@ define('diagramStore', ['jquery', 'xwiki-meta', 'draw.io', 'xwiki-events-bridge'
     oldImportFiles.apply(this, newArgs);
   }
 
+  //
+  // Change the URL generated by export for containing the actual host.
+  //
+  var oldCreateLink = EditorUi.prototype.createLink;
+  EditorUi.prototype.createLink = function() {
+    let rawURL = oldCreateLink.apply(this, arguments);
+    // Do not include '#' because it's automatically added by getURL function below.
+    let documentFragment = rawURL.substring(rawURL.indexOf('#') + 1);
+    if (documentFragment.substring(0, 1) == 'R') {
+      let queryString = '';
+      if (rawURL.indexOf('/?') &gt; -1) {
+        queryString = rawURL.substring (rawURL.indexOf('/?'), rawURL.indexOf('#R'));
```

Comment:
```suggestion
        queryString = rawURL.substring (rawURL.indexOf('?') + 1, rawURL.indexOf('#R'));
```

I think you need to skip the '?' because it is included by [getURL()](https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-8.4/xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/xwiki.js#L1375).

Document fragment index is also computed twice so could be worth saving it.